src/components/NavBar/MenuItemMobile/index.tsx

```diff
@@ -0,0 +1,184 @@
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  Button,
+  createStyles,
+  Grid,
+  Switch,
+  Theme,
+  Typography,
+  withStyles,
+  WithStyles,
+} from '@material-ui/core';
+import Accordion from '@material-ui/core/Accordion';
+import AccordionSummary from '@material-ui/core/AccordionSummary';
+import AccordionDetails from '@material-ui/core/AccordionDetails';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCaretDown } from '@fortawesome/free-solid-svg-icons';
+
+import { LayerType, TableType } from '../../../config/types';
+import { addLayer, removeLayer } from '../../../context/mapStateSlice';
+import { loadTable } from '../../../context/tableStateSlice';
+import { layersSelector } from '../../../context/mapStateSlice/selectors';
+import { menuList } from '../utils';
+
+function MenuItemMobile({ classes }: MenuItemMobileProps) {
+  const [expanded, setExpanded] = useState<string>('');
+
+  const selectedLayers = useSelector(layersSelector);
+
+  const dispatch = useDispatch();
+
+  const handleChange = (panel: string) => (
+    event: object,
+    newExpanded: boolean,
+  ) => {
+    setExpanded(newExpanded ? panel : '');
+  };
+
+  const toggleLayerValue = (prevChecked: boolean, layer: LayerType) => {
+    if (prevChecked) {
+      dispatch(removeLayer(layer));
+    } else {
+      dispatch(addLayer(layer));
+    }
+  };
+
+  const showTableClicked = (table: TableType) => {
+    dispatch(loadTable(table.id));
+  };
+
+  return (
+    <>
+      {menuList.map(({ title, icon, layersCategories }) => (
+        <Accordion
+          className={classes.accordion}
+          key={title}
+          square
+          expanded={expanded === title}
+          onChange={handleChange(title)}
+        >
+          <AccordionSummary
+            className={classes.accordionSummary}
+            expandIcon={<FontAwesomeIcon icon={faCaretDown} />}
+            IconButtonProps={{ color: 'inherit', size: 'small' }}
+            aria-controls={title}
+            id={title}
+          >
+            <img className={classes.icon} src={icon} alt={title} />
+            <Typography variant="body2">{title}</Typography>
+          </AccordionSummary>
+          <AccordionDetails className={classes.accordionDetail}>
+            <Grid container direction="column">
+              {layersCategories.map(
+                ({ title: categoryTitle, layers, tables }) => (
+                  <Grid
+                    item
+                    key={categoryTitle}
+                    className={classes.categoryContainer}
+                  >
+                    <Typography
+                      variant="body2"
+                      className={classes.categoryTitle}
+                    >
+                      {categoryTitle}
+                    </Typography>
+                    <hr />
+
+                    {layers.map(layer => {
+                      const { id: layerId, title: layerTitle } = layer;
```

Comment:
can we move this to the function args? If there's a specific reason its a new line, all good :+1:

Reply:
This one actually I followed the codes from `<MenuItem/>`, but if I see again the `layer` argument is used in the bottom. How about we remove one line `const { id: layerId, title: layerTitle } = layer;` and use `layer.id` and `layer.title` directly instead?

Reply:
For the comment I mentioned, we can leave as it is :+1:

On another note, I didn't realise how much of the `MenuItem` was duplicated in `MenuItemMobile`. We should really cut this down with helper components or something else so the code is more maintainable. For example, the switch and table button code are almost identical.

Reply:
I separated the switch/table so that it can be reused by `MenuItem` and `MenuItemMobile`. kindly check in my new commit 🙏